lightning/restore/restore.go

```diff
@@ -1400,19 +1404,41 @@ func (t *TableRestore) initializeColumns(columns []string, ccp *ChunkCheckpoint)
 			colPerm = append(colPerm, -1)
 		}
 	} else {
-		colPerm = t.parseColumnPermutations(columns)
+		var err error
+		colPerm, err = t.parseColumnPermutations(columns)
+		if err != nil {
+			return errors.Trace(err)
+		}
 	}
 
 	ccp.ColumnPermutation = colPerm
+	return nil
 }
 
-func (t *TableRestore) parseColumnPermutations(columns []string) []int {
+func (t *TableRestore) parseColumnPermutations(columns []string) ([]int, error) {
 	colPerm := make([]int, 0, len(t.tableInfo.Core.Columns)+1)
 
 	columnMap := make(map[string]int)
 	for i, column := range columns {
 		columnMap[column] = i
 	}
+
+	tableColumnMap := make(map[string]int)
+	for i, col := range t.tableInfo.Core.Columns {
+		tableColumnMap[col.Name.L] = i
+	}
+
+	// check if there are some unknown columns
+	var unknownCols []string
+	for _, c := range columns {
+		if _, ok := tableColumnMap[c]; !ok && c != model.ExtraHandleName.L {
+			unknownCols = append(unknownCols, c)
+		}
+	}
+	if len(unknownCols) > 0 {
+		return colPerm, errors.Errorf("unknown columns in csv header %s", unknownCols)
```

Comment:
```suggestion
		return colPerm, errors.Errorf("unknown columns in header: %s", unknownCols)
```

this function is not CSV-specific